addons/onboarding/controllers/onboarding.py

```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.http import request
+
+
+class OnboardingController(http.Controller):
+    @http.route(['/onboarding/<int:onboarding_id>',
+                 '/onboarding/<string:onboarding_alias>'], auth='user', type='json')
+    def get_onboarding_data(self, onboarding_id=None, onboarding_alias=None):
+        if not request.env['onboarding.onboarding'].check_access_rights('read', raise_exception=False):
+            return {}
+        onboarding = request.env['onboarding.onboarding'].search([
+            ('id', '=', onboarding_id) if onboarding_id else ('onboarding_alias', '=', onboarding_alias)
+        ])
+
+        if not onboarding or onboarding.search_or_create_progress().current_progress_id.is_onboarding_closed:
+            return {}
+
+        return {
+            'html': request.env['ir.qweb']._render(
```

Comment:
Maybe add a note telling this is the standard for JS of onboardings, could try to cleanup that once we have a bit of time